File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/data/storage/SettingsRepositoryImpl.java
package space.lopatkin.spb.helpboardgamecard.data.storage;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import space.lopatkin.spb.helpboardgamecard.data.repository.SettingsRepository;

public class SettingsRepositoryImpl implements SettingsRepository {
    private final static String APPLICATION_PREFERENCES = SettingsRepositoryImpl.class.getSimpleName() + "APPLICATION_PREFERENCES";
    private final static String KEYBOARD_VARIANT = "KEYBOARD_VARIANT";

    private Context context;
    private SharedPreferences preferences;

    public SettingsRepositoryImpl(Application application) {
        this.context = application.getApplicationContext();
        preferences = context.getSharedPreferences(APPLICATION_PREFERENCES, Context.MODE_PRIVATE);
    }

    @Override
    public void saveKeyboardVariant(int keyboardVariant) {
        preferences
                .edit()
                .putInt(KEYBOARD_VARIANT, keyboardVariant)
                .apply();
    }

    @Override
    public int getKeyboardVariant() {
        return preferences
                .getInt(KEYBOARD_VARIANT, -1);
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/data/storage/preferences/SettingsRepositoryImpl.kt
package space.lopatkin.spb.helpboardgamecard.data.storage.preferences

import android.content.Context
import android.content.SharedPreferences
import space.lopatkin.spb.helpboardgamecard.data.storage.repository.SettingsRepository

private const val APPLICATION_PREFERENCES = "APPLICATION_PREFERENCES"
private const val KEYBOARD_TYPE = "KEYBOARD_TYPE"

class SettingsRepositoryImpl(private val context: Context) : SettingsRepository {

    private val preferences: SharedPreferences = context.getSharedPreferences(APPLICATION_PREFERENCES, Context.MODE_PRIVATE)

    override fun saveKeyboardType(type: Int) {
        preferences
            .edit()
            .putInt(KEYBOARD_TYPE, type)
            .apply()
    }

    override fun getKeyboardType(): Int {
        return preferences
            .getInt(KEYBOARD_TYPE, -1)
    }

}
